extensions/resteasy-reactive/rest-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProcessor.java

```diff
@@ -163,7 +164,9 @@ AdditionalBeanBuildItem beans() {
                 // just because it's a bean.
                 // Whether it is used in RESTEasy Reactive is determined elsewhere
                 .addBeanClass(FullyFeaturedServerJacksonMessageBodyWriter.class)
-                .setUnremovable().build();
+                .setDefaultScope(DotNames.APPLICATION_SCOPED)
```

Comment:
I wonder if `@Singleton` would be a better match :thinking:. You would not need the no-args constructors and it should be slightly better perf-wise.

Reply:
That would mean that all dependencies would be created too, right?

Reply:
Yes, but that's what happens with `@Dependent` right now... Nevertheless, `@ApplicationScoped` should be fine too ;-).

Reply:
Let's go with `@ApplicationScoped` then :)